grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/CorenessQueryImpl.java

```diff
@@ -95,32 +86,28 @@
                     new DegreeDistributionMapReduce(ofLabelIds, CorenessVertexProgram.CORENESS),
                     subLabelIds);
         } catch (NoResultException e) {
-            LOGGER.info("Coreness query is finished in " + (System.currentTimeMillis() - startTime) + " ms");
+            LOGGER.info("Coreness query finished in " + (System.currentTimeMillis() - startTime) + " ms");
             return Collections.emptyMap();
         }
 
-        LOGGER.info("Coreness query is finished in " + (System.currentTimeMillis() - startTime) + " ms");
+        LOGGER.info("Coreness query finished in " + (System.currentTimeMillis() - startTime) + " ms");
         return result.memory().get(DegreeDistributionMapReduce.class.getName());
     }
 
     @Override
-    public CorenessQuery minK(int k) {
+    public CorenessQuery minK(long k) {
         this.k = k;
         return this;
     }
 
     @Override
     public CorenessQuery of(String... ofTypeLabels) {
-        return of(Arrays.stream(ofTypeLabels).map(Label::of).collect(Collectors.toSet()));
+        return (CorenessQuery) super.setOfTypeLabels(ofTypeLabels);
```

Comment:
Could this sort of casting be solved by having the super method return a generic?

Reply:
I have removed this method using generic. Please have a look.